lib/gui/app/pages/main/MainPage.tsx

```diff
@@ -188,12 +197,17 @@ export class MainPage extends React.Component<
 						/>
 					)}
 
-					<div
+					<Flex
 						className="page-main row around-xs"
 						style={{ margin: '110px 50px' }}
 					>
 						<div className="col-xs">
-							<ImageSelector flashing={this.state.isFlashing} />
+							<SourceSelector
+								flashing={this.state.isFlashing}
+								afterSelected={(source: SourceOptions) =>
```

Comment:
Please throw an error if we're using `Http` and `source.imagePath` does not start with `http://` or `https://`
Something like "Only http:// and https:// protocols are supported"

Reply:
Added after checking for supported image formats too